joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/astcreation/AstForDeclarationsCreator.scala

```diff
@@ -45,6 +46,47 @@ trait AstForDeclarationsCreator(implicit withSchemaValidation: ValidationMode) {
     Seq(typeDeclAst)
   }
 
+  protected def astForRecordDeclaration(recordDecl: DotNetNodeInfo): Seq[Ast] = {
+    val name     = nameFromNode(recordDecl)
+    val fullName = astFullName(recordDecl)
+    val typeDecl = typeDeclNode(recordDecl, name, fullName, relativeFileName, code(recordDecl))
+    scope.pushNewScope(TypeScope(fullName))
+    val modifiers         = astForModifiers(recordDecl)
+    val paramList         = recordDecl.json(ParserKeys.ParameterList)
+    var membersFromParams = Seq[Ast]()
+
+    // Covers the case where record type can be declared as `record Person(string Name);`
+    // Here, Person should be a TypeDecl and Name should be a member instead of a parameter
+    if (!paramList.isNull) {
+      membersFromParams = paramList(ParserKeys.Parameters).arr
+        .map(p => {
+          val metaData     = p(ParserKeys.MetaData)
```

Comment:
This should use `createDotNetNodeInfo` as it extracts all of the meta data information and sets the type for you.
You should the be able to do something like
```scala
paramList(ParserKeys.Parameters).arr
  .map(createDotNetNodeInfo)
  .map { paramNode => 
      val name = nameFromNode(paramNode) // the helper function for extracting identifier info
      val typeFullName = nodeTypeFullName(paramNode) // helper function for obtaining type info (uses type map)
      memberNode(paramNode, name, ... etc)
  }.toSeq
```

Reply:
Good point